docs(DR-481): document k8s release agent 1.6.0 dynamic managed namespaces (#10405)
circleci/circleci-docs: docs/guides/modules/deploy/pages/set-up-the-circleci-release-agent.adoc

. Navigate to the **Environments** tab.
. Select the gear icon icon:settings[Settings icon] corresponding to the environment you just created to enter the Environment Integration Settings page.
. Select btn:[Create Integration Token] and then btn:[Create Token].
. The new token will be displayed. Copy the token, then select btn:[Done].

> The token is shown only once and cannot be retrieved from CircleCI later. Store it in your organization's secret manager. Examples include HashiCorp Vault, AWS Secrets Manager, or a CircleCI context variable.

Reviewer: ```suggestion
*The integration token is shown only once and cannot be retrieved from CircleCI later*. Store the integration token in your organization's secret manager. Examples of secret managers include HashiCorp Vault, AWS Secrets Manager, or a CircleCI context variable.
```

Pass the token to Helm at install or upgrade time. Use an environment variable (for example, `++--set tokenSecret.token="${CCI_INTEGRATION_TOKEN}"++`) or a values file that is not committed to source control.

> After installation, the chart stores the token in a Kubernetes Secret in the `circleci-release-agent-system` namespace. On subsequent `helm upgrade` runs, you do not need to supply the token again unless you use `--reset-values`. The same applies if you apply a values file that omits `tokenSecret.token` without `--reuse-values`.

Reviewer: ```suggestion
After installation, the Helm chart stores the integration token in a Kubernetes Secret in the `circleci-release-agent-system` namespace. On subsequent `helm upgrade` runs, you do not need to supply the integration token again unless you use `--reset-values`. The same applies if you apply a values file that omits `tokenSecret.token` without also setting `--reuse-values`.
```

[#install-the-release-agent]
=== 3. Install the release agent

Use the following command to install the Helm chart to integrate CircleCI's release agent into your Kubernetes cluster.

> If required, you can include a list of Kubernetes namespaces you want to monitor. If you only want to monitor the `default` namespace, you do not need to supply a list with `--set managedNamespaces="{namespace1,namespace2}"`. The `default` namespace is monitored by default. If you struggle to pass the `managedNamespaces` value, try enclosing the array value or the whole parameter itself in a string, with double or single quotes.

Reviewer: ```suggestion
If required, you can include a list of Kubernetes namespaces you want to monitor with `--set managedNamespaces="{namespace1,namespace2}"`. If you only want to monitor the `default` namespace, you do not need to supply a list. The `default` namespace is monitored by default. If you struggle to pass the `managedNamespaces` value, try enclosing the array value or the whole parameter itself in a string, with double or single quotes.
```

. Select the **Environments** tab.
. Select the name of your environment integration to view the environment details page.

[#managed-namespaces-mode]
=== Configure managed namespaces mode

> Release agent v1.6.0 and Helm chart v1.6.0 introduce a `managedNamespacesMode` Helm value. This setting (exposed as `MANAGED_NAMESPACES_MODE` on the agent deployment) controls how the agent determines which namespaces to monitor:

Reviewer: ```suggestion

NOTE: Release agent v1.6.0 and Helm chart v1.6.0 or higher are required to use the manage namespaces mode.

Release agent v1.6.0 and Helm chart v1.6.0 introduce a `managedNamespacesMode` Helm value. This setting (exposed as `MANAGED_NAMESPACES_MODE` on the agent deployment) controls how the agent determines which namespaces to monitor. Managed namespaces mode has two options as follows:
```

* `static` (default) -- the agent monitors the exact namespace names listed in `managedNamespaces`. This is the same behavior as prior versions and requires no additional RBAC permissions.

> * `dynamic` -- the agent discovers namespaces at runtime by matching the patterns in `managedNamespaces` against all cluster namespaces. Use glob patterns such as `team-*` to monitor any namespace whose name matches the pattern.

Reviewer: ```suggestion
* `static` (default): The agent monitors the exact namespace names listed in `managedNamespaces`. This is the same behavior as prior versions and requires no additional RBAC permissions.
* `dynamic`: The agent discovers namespaces at runtime by matching the patterns in `managedNamespaces` against all cluster namespaces. Use glob patterns such as `team-*` to monitor any namespace whose name matches the pattern.
```

[#static-mode]
==== Use static mode (default)

In static mode, `managedNamespaces` contains a list of exact namespace names. This mode requires no additional RBAC permissions beyond those granted in earlier chart versions.

> 

Reviewer: ```suggestion

To upgrade and use `static` mode, run the following command:

```

[source,shell]
----
helm upgrade --install circleci-release-agent-system release-agent/circleci-release-agent \
  --set tokenSecret.token=[YOUR_CCI_INTEGRATION_TOKEN] --create-namespace \
  --namespace circleci-release-agent-system \
  --set managedNamespaces="{namespace1,namespace2}"
----

[#dynamic-mode]
==== Use dynamic mode

In dynamic mode, `managedNamespaces` accepts glob patterns. The agent queries the Kubernetes API at runtime to resolve which namespaces match each pattern.

> When installing or upgrading for dynamic mode, set `rbac.clusterRoles=true` (in addition to `rbac.create=true`). This enables the chart to create the required `ClusterRole` with `get`, `list`, and `watch` permissions on the `namespaces` resource.

Reviewer: ```suggestion
When installing or upgrading for dynamic mode, set `rbac.clusterRoles=true`, in addition to `rbac.create=true`. This enables the chart to create the required `ClusterRole` with `get`, `list`, and `watch` permissions on the `namespaces` resource.
```

> 

Reviewer: ```suggestion

To upgrade and use `dynamic` mode, run the following command:

```

[source,shell]
----
helm upgrade --install circleci-release-agent-system release-agent/circleci-release-agent \
  --set tokenSecret.token=[YOUR_CCI_INTEGRATION_TOKEN] --create-namespace \
  --namespace circleci-release-agent-system \
  --set managedNamespaces="{team-*,platform-*}" \
  --set managedNamespacesMode=dynamic \
  --set rbac.create=true \
  --set rbac.clusterRoles=true
----

NOTE: The `rbac.clusterRoles=true` flag also enables the RBAC permissions the agent needs to watch workloads across all matching namespaces. Both flags are required for dynamic mode to function correctly.

[#upgrade-path]
==== Upgrade from an earlier version

> If you are upgrading from a version earlier than 1.6.0, your existing configuration uses static mode by default. No action is required — the agent continues to monitor the exact namespace names already listed in `managedNamespaces`, and no new RBAC permissions are granted.

Reviewer: ```suggestion
If you are upgrading from a version earlier than 1.6.0, your existing configuration uses static mode by default. No action is required. The agent continues to monitor the exact namespace names already listed in `managedNamespaces`, and no new RBAC permissions are granted.
```

----

[#dynamic-mode]
==== Use dynamic mode

In dynamic mode, `managedNamespaces` accepts glob patterns. The agent queries the Kubernetes API at runtime to resolve which namespaces match each pattern.

When installing or upgrading for dynamic mode, set `rbac.clusterRoles=true`, in addition to `rbac.create=true`. This enables the chart to create the required `ClusterRole` with `get`, `list`, and `watch` permissions on the `namespaces` resource.

To upgrade and use `dynamic` mode, run the following command:

[source,shell]
----
helm upgrade --install circleci-release-agent-system release-agent/circleci-release-agent \
  --set tokenSecret.token=[YOUR_CCI_INTEGRATION_TOKEN] --create-namespace \
  --namespace circleci-release-agent-system \
  --set managedNamespaces="{team-*,platform-*}" \
  --set managedNamespacesMode=dynamic \
  --set rbac.create=true \
  --set rbac.clusterRoles=true
----

NOTE: The `rbac.clusterRoles=true` flag also enables the RBAC permissions the agent needs to watch workloads across all matching namespaces. Both flags are required for dynamic mode to function correctly.

[#upgrade-path]
==== Upgrade from an earlier version

> If you are upgrading from a version earlier than 1.6.0, your existing configuration uses static mode by default. No action is required. The agent continues to monitor the exact namespace names already listed in `managedNamespaces`, and no new RBAC permissions are granted.

Reviewer: How do people know/check what version they are currently using?

Author: Added some info about it.
